csharp/test/Ice/proxy/AllTests.cs

```diff
@@ -16,8 +16,64 @@ public static IMyClassPrx allTests(TestHelper helper)
             Communicator? communicator = helper.Communicator();
             TestHelper.Assert(communicator != null);
             System.IO.TextWriter output = helper.GetWriter();
-            output.Write("testing stringToProxy... ");
+            output.Write("testing proxy parsing (URI)... ");
             output.Flush();
+
+            string[] uriStringArray =
+            {
+                "ice+tcp://host.zeroc.com/identity#facet",
+                "ice+tcp://host.zeroc.com:1000/category/name",
+                "ice+tcp://host.zeroc.com/category/name%20with%20space",
+                "ice+ws://host.zeroc.com//identity",
+                "ice+ws://host.zeroc.com//identity?alt-endpoints=host2.zeroc.com",
+                "ice+ws://host.zeroc.com//identity?alt-endpoints=host2.zeroc.com:10000",
+                "ice+ws://host.zeroc.com//identity?protocol=ice1&source-address=10.1.2.4&compress=true",
+                "ice+ws://host.zeroc.com//identity?protocol=ice1&alt-endpoints=host2.zeroc.com?" +
+                    "source-address=10.1.2.4$compress=true",
+                "ice+tcp://[::1]:10000/identity?alt-endpoints=host1:10000,host2,host3,host4",
+                "ice:location//identity#facet",
+                "ice+tcp://host.zeroc.com//identity",
+                "ice+tcp://host.zeroc.com:/identity", // another syntax for empty port
+                "ice+universal://com.zeroc.ice/identity?transport=iaps&options=a,b%2Cb,c,d",
+                "ice+universal://host.zeroc.com/identity?transport=100",
+                "ice+universal://[ab:cd:ef:00]/identity?transport=bt",
+                "ice+udp://[::1]/test?source-address=::1&interface=0:0:0:0:0:0:0:1%25lo", // unescaped as ..:1%lo
+                "ice+ws://host.zeroc.com/identity?resource=/foo%2Fbar?/xyz",
+            };
+
+            foreach (string uriString in uriStringArray)
+            {
+                _ = IObjectPrx.Parse(uriString, communicator);
+                // output.WriteLine($"{uriString} = {prx}");
+            }
+
+            string[] badUriStringArray =
+            {
+                "ice+tcp://host.zeroc.com:foo",     // missing host
+                "ice+tcp:identity?protocol=invalid", // invalid protocol
+                "ice+universal://host.zeroc.com", // missing transport
+                "ice+universal://host.zeroc.com?transport=100&protocol=ice1", // invalid protocol
+                "ice://host:1000/identity", // host
```

Comment:
What is bad about this URI?

Reply:
Ok, It's because it's indirect, maybe update the comment?

Reply:
Updated.